### PR TITLE
Edit Event Form

### PR DESCRIPTION
**Pull request by OdeChan**
Added a form to edit the event details
Added navigation tabs as base for event features
Created eventedit endpoint to render the form page

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -30,3 +33,8 @@ def profile(request: Request):
         "username": current_username,
         "events": upcouming_events
     })
+
+
+@app.get("/eventedit")
+def eventedit(request: Request):
```
Comment (ivarshav) [inline app/main.py:39]: Maybe better to place in a `routes/evnets.py` file.

File: tests/test_app.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from fastapi.testclient import TestClient
+from app.main import app
+
+client = TestClient(app)
+
+def test_eventedit():
```
Comment (ivarshav) [inline tests/test_app.py:6]: Also, better to place in `test_events.py` file.

File: app/main.py [hunk 3]
```diff
@@ -1,13 +1,17 @@
...
+
 from fastapi import FastAPI, Request
 from fastapi.staticfiles import StaticFiles
 from fastapi.templating import Jinja2Templates
 
-
 app = FastAPI()
 
-app.mount("/static", StaticFiles(directory="static"), name="static")
+app_path = os.path.dirname(os.path.realpath(__file__))
```
Comment (ivarshav) [inline app/main.py:9]: Not sure I understand why you need to add that in order to run the tests?

File: app/static/eventedit.css [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+.form_row,
+.form_row_start,
+.form_row_end {
+	display: flex
+}
+
+.form_row {
+	flex: 1;
+	min-height: 35px;
```
Comment (yammesicka) [inline app/static/eventedit.css:37]: Use `em` please :)

Try to avoid using `px`

File: app/main.py (hunk 1 above)
Comment (Gonzom) [reply app/main.py:39]: This doesn't seem to be correct semantic usage of a get request and should instead be: @app.put("/event").

File: app/templates/eventdetailsformtab.html [hunk 5]
```diff
@@ -0,0 +1,48 @@
+<div class="form_row">
```
Comment (yammesicka) [inline app/templates/eventdetailsformtab.html:1]: Change the name to event_details_form_tab.html

File: app/main.py [hunk 6]
```diff
@@ -30,3 +33,8 @@ def profile(request: Request):
         "username": current_username,
         "events": upcouming_events
     })
+
+
+@app.get("/eventedit")
```
Comment (yammesicka) [inline app/main.py:38]: Maybe `/event/edit`?

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 6 above)
Comment (Gonzom) [reply app/main.py:38]: not @app.put("/event")?

File: app/main.py (hunk 3 above)
Comment (OdeChan) [reply app/main.py:9]: The tests folder currently resides outside the app folder, so when running pytest, app.mount looks for the "static" folder in the current working directory - tests. This fix looks for the static folder path always in the same directory as main.py

File: app/main.py (hunk 6 above)
Comment (OdeChan) [reply app/main.py:38]: @Gonzom this is specific for the get method, not post or put

File: app/templates/eventdetailsformtab.html (hunk 5 above)
Comment (OdeChan) [reply app/templates/eventdetailsformtab.html:1]: Will fix :)

File: app/main.py (hunk 6 above)
Comment (Gonzom) [reply app/main.py:38]: Ah, the get request is to reach the edit page, not to edit the event. My mistake!

**Comment by yammesicka** [comment]
Please change the PR to merge into PythonFreeCourse:develop

File: app/main.py (hunk 6 above)
Comment (OdeChan) [reply app/main.py:38]: @yammesicka 👍

**Pull request merged**